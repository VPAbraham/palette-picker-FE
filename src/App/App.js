import React, { Component } from 'react';
import Modal from 'react-modal';
import './App.scss';
import Nav from '../Nav/Nav';
import Palette from '../Palette/Palette';
import PaletteForm from '../PaletteForm/PaletteForm';
import ProjectForm from '../ProjectForm/ProjectForm';
import { getColors } from '../apiCalls/apiCallsColors';
import whiteClose from '../assets/images/close_white.svg';
import { getProjects, getPalettes } from '../apiCalls/apiCalls';

class App extends Component {
  constructor() {
    super()
    this.state = {
      color1: '#baddc3',
      color2: '#ff6767',
      color3: '#ffe596',
      color4: '#b3e6ff',
      color5: '#41406d',
      modalIsOpen: false,
      palettes: [],
      projects: [],
      creatingProj: false
    }
    this.openModal = this.openModal.bind(this);
    this.closeModal = this.closeModal.bind(this);
  }

  async componentDidMount() {
    const newPalette = await getColors();
    this.setState({
      randomizedPalette: newPalette
    });
    const allProjects = await getProjects();
    this.setState({
      projects: allProjects
    });
    const allPalettes = await getPalettes();
    this.setState({
      palettes: allPalettes
    });
  }

  refreshColors = async () => {
    const newPalette = await getColors();
    this.setState({
      color1: newPalette[0],
      color2: newPalette[1],
      color3: newPalette[2],
      color4: newPalette[3],
      color5: newPalette[4]
    });
  };

  refreshPalettes = (newPalette) => {
    const allPalettes = this.state.palettes;
    if(newPalette.name) {
      allPalettes.push(newPalette);
      this.setState({palettes: allPalettes})
    }
  }

  openModal() {
    this.setState({ modalIsOpen: true });
  };

  closeModal() {
    this.setState({ modalIsOpen: false });
  };

  selectPalette = (color1, color2, color3, color4, color5) => {
    this.setState({
      color1: color1,
      color2: color2,
      color3: color3,
      color4: color4,
      color5: color5
    });
  };

  render() {
    return(
      <div className="App" id="root">
        <Nav
          openModal={this.openModal}
          refreshColors={this.refreshColors}
        />
        <Modal
          isOpen={this.state.modalIsOpen}
          onRequestClose={this.closeModal}
          className="Modal"
          contentLabel="Example Modal"
          key="Modal"
        >
          <img src={whiteClose} onClick={this.closeModal} alt="white x symbol" className="modalClose"/>
          <section className="menu">

            <PaletteForm
              color1={this.state.color1}
              color2={this.state.color2}
              color3={this.state.color3}
              color4={this.state.color4}
              color5={this.state.color5}
              projects={this.state.projects}
              refreshPalettes={this.refreshPalettes}
              key="PaletteForm"
            />
            <ProjectForm
              palettes={this.state.palettes}
              projects={this.state.projects}
              selectPalette={this.selectPalette}
              key="ProjectForm"
            />
          </section>
        </Modal>
        <main>
          <Palette
            color1={this.state.color1}
            color2={this.state.color2}
            color3={this.state.color3}
            color4={this.state.color4}
            color5={this.state.color5}
            key="palette"
          />
        </main>
      </div>
    );
  }
}

export default App;
