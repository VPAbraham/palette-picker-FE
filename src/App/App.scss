@import url('https://fonts.googleapis.com/css?family=Montserrat&display=swap');

* {
  box-sizing: border-box;
  margin: 0;
}

html {
  background: rgb(242, 242, 242);
}

.App {
  text-align: center;

  main {
    position: absolute;
    left: 0;
    right: 0;
    top: 20vh;
  }
}

.Modal {
  background-color: #000000;
  padding: 8%;
  height: 100%;
  overflow-y: scroll;
}

.modalClose {
  float: right;
  width: 3em;
}

h1 {
  color: rgb(189, 189, 189);
  display: flex;
  font-family: 'Montserrat', sans-serif;
  font-size: 1.5em;
  letter-spacing: 15px;
  line-height: 55px;
  text-align: left;
}

.menu {
  display: flex;
  flex-direction: column;

  .proj-name-input {
    display: flex;
    flex-direction: column;
    width: 75%;

    input {
      width: 100%;
      transition: .6s;
    }

    button {
      font-size: 16px;
      padding: 1em;
      width: 25%;
      margin: 3em 0 2em 0;
    }
  }
}

.menu-items {
  display: flex;
  flex-direction: row;
}

.modalPlus {
  margin-right: 24px;
  width: 1.5em;
}