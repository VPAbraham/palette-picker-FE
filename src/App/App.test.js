import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('App', () => {
  it('should match the snapshot with all data passed in', () => {
    let wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot();
  });
});