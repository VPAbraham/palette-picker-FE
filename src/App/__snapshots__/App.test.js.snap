// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in 1`] = `
<div
  className="App"
  id="root"
>
  <Nav
    openModal={[Function]}
    refreshColors={[Function]}
  />
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="Modal"
    closeTimeoutMS={0}
    contentLabel="Example Modal"
    isOpen={false}
    key="Modal"
    onRequestClose={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <img
      alt="white x symbol"
      className="modalClose"
      onClick={[Function]}
      src="close_white.svg"
    />
    <section
      className="menu"
    >
      <PaletteForm
        color1="#baddc3"
        color2="#ff6767"
        color3="#ffe596"
        color4="#b3e6ff"
        color5="#41406d"
        key="palette-form"
        projects={Array []}
        refreshPalettes={[Function]}
      />
      <ProjectForm
        hover={false}
        key="ProjectForm"
        palettes={Array []}
        projects={Array []}
        selectPalette={[Function]}
      />
    </section>
  </Modal>
  <main>
    <Palette
      color1="#baddc3"
      color2="#ff6767"
      color3="#ffe596"
      color4="#b3e6ff"
      color5="#41406d"
      key="palette"
    />
  </main>
</div>
`;
